### DISCUSSION TITLE
Array filter types with classes extends

### DISCUSSION THREAD

**Discussion by JoseLuisRNP** [mattpocock/ts-reset#122] [Q&A]
Can we extends de Array.filter(Boolean) funcionality of ts-error solving this problem? When we have a array of elements extendings from a class and we want filter only certain instanceof specific class we need to use always a predicate.

https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAcgAIwjAhgGwLQwJ5gCmAzgMZTBgwD0Mx2UJhMCA3AFDvoBGxMU6UvFKZ0xYnABCYwnADe7OErjAAJgC44AOwCuIboSgdFy0hC18oOodAAUASnknlymAAtgxAHRq4AXjgAWXR3LwEtVQgQBw4XAF92BM4RMQlpKDhCAA8YQgi0mScXfCJ-RG50KDZE5NFxOAAxCAhMnLzVAuJZBWKCWQCEADNm6qSzC3hCTEIQPLoygG0tQgB3RuaHABptValKrZ21pogHAF1jcb44YYgJAKmZue9B4ExcqFtbKcc-AD5MzAqCboLSkQgQQbrCD2DiXeA3YgABUYqmApBChEIqjKD1mWjoXheb0Mn2+mimKgkx38-wpwGBoPBkOOMPYcOuzWIAHE8oY0TjpniCUT3gAeY6-UmYH60wH0vggsEQqEwoA

**Comment by mattpocock** [comment]
Unfortunately not!
